### ISSUE TITLE
Adding the SP.RelatedItemManager api in the framework

### ISSUE THREAD

**Issue by ravenboilinux** [SharePoint/PnP-JS-Core#397]
#### Category
[X ] Enhancement

[ ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [   2.0.2      ]

#### Expected / Desired Behavior / Question
I work with workflow task list alot. I mostly like to post back to the SP.RelatedItemManager to get the list item that is related to the workflow item. 

it would be nice to do something like this
pnp.sp.relateditem({'SourceListName': 'Workflow Tasks','SourceItemID': 12})
and that would return something like this json item back.

**Comment by koltyakov** [comment]
Hi,

Just shoot you with some info to think out, maybe it can be useful for the feature design.

SP.ReletedItemManager has a few of methods and corresponding REST endpoints:
- `_api/SP.RelatedItemManager.GetRelatedItems`
- `_api/SP.RelatedItemManager.AddSingleLinkToUrl`
- `_api/SP.RelatedItemManager.DeleteSingleLink`
- ... and some more

I would say, if to implement `RelatedItemManager` then to implement it in a fluent way, like:

```javascript
pnp.sp.web.relatedItemManager.getRelatedItems(...getRelatedItemsArgs)
  .then(console.log).catch(console.log);

pnp.sp.web.relatedItemManager.addSingleLinkToUrl(...addSingleLinkToUrlArgs)
  .then(console.log).catch(console.log);

pnp.sp.web.relatedItemManager.deleteSingleLink(...deleteSingleLinkArgs)
  .then(console.log).catch(console.log);

/**
 * And so on...
 * ... addSingleLink
 * ... addSingleLinkFromUrl
 * ... getPageOneRelatedItems
 */
```

Maybe there should be some 'fluency' for arguments but, from the first glance, they seemed to be alright as a typed object or typed arguments.

**Comment by ravenboilinux** [comment]
That does sound like a good idea. Just one thing I would change is:
```javascript
pnp.sp.relatedItemManager.getRelatedItems(...getRelatedItemsArgs)
  .then(console.log).catch(console.log);

pnp.sp.relatedItemManager.addSingleLinkToUrl(...addSingleLinkToUrlArgs)
  .then(console.log).catch(console.log);

pnp.sp.relatedItemManager.deleteSingleLink(...deleteSingleLinkArgs)
  .then(console.log).catch(console.log);
'''
instead of 
```javascript
pnp.sp.web.relatedItemManager.getRelatedItems(...getRelatedItemsArgs)
  .then(console.log).catch(console.log);

pnp.sp.web.relatedItemManager.addSingleLinkToUrl(...addSingleLinkToUrlArgs)
  .then(console.log).catch(console.log);

pnp.sp.web.relatedItemManager.deleteSingleLink(...deleteSingleLinkArgs)
  .then(console.log).catch(console.log);
'''

This will allow it to be easier to implement because the base url is the site collection url plus the '_api'

**Comment by patrick-rodgers** [comment]
@ravenboilinux & @koltyakov - thanks for the idea I'll have a look. I am asking because I don't know - is this ever scoped to a web? Meaning does a web have a related item manager or is it always at the site collection level regardless of sub web location?

**Comment by ravenboilinux** [comment]
is this ever scoped to a web?
It is scoped to the base url 
http://ge.guardian.com/automotiveeurope/_api/SP.RelatedItemManager.GetRelatedItems
That rest call will only look at the list from  site that is the base url.
There is no SP.RelatedItemManager under /web/ or /site/

**Comment by patrick-rodgers** [comment]
https://sharepointfieldnotes.blogspot.com/2014/12/managing-related-items-with-sharepoint.html

**Comment by ravenboilinux** [comment]
That is how you do it with C#. Is similar to do the same with Javascript with ajex too.

**Comment by ravenboilinux** [comment]
It is similar in javascript too.

**Comment by patrick-rodgers** [comment]
That is just me taking notes. I am working on finishing up sharing and debating trying to get this in for Friday's release. Largely dependent on my other work at this point.

**Comment by patrick-rodgers** [comment]
Ok, going to give a shot to adding this in time for 2.0.3 - no guarantees but just flagged it as in progress.

**Comment by patrick-rodgers** [comment]
Interestingly this is scoped to the web level and is not static at the root. So these two get the list in the 
scope of the web:

https://site.sharepoint.com/sites/dev/_api/SP.RelatedItemManager

https://site.sharepoint.com/sites/dev/0d54812e-153b-4da3-a676-99ff84ab2b77/_api/SP.RelatedItemManager

**Comment by patrick-rodgers** [comment]
Added support for this in this [PR](#407) and it will be included in 2.0.3. It will chain off web and use the relatedItems property. Associated [wiki page](https://github.com/SharePoint/PnP-JS-Core/wiki/Working-With:-Related-Items).

Closing as added.
